### DISCUSSION TITLE
✨ Dockerignore syntax.

### DISCUSSION THREAD

**Discussion by stobias123** [dagger/dagger#12082] [Ideas]
### What are you trying to do?

The new `WithoutDirectory` syntax is great, but I am looking for something closer to `.dockerignore` syntax.

I don't know the name of all directories I'd like to ignore when I read the host dir, but I know they will all contain a folder `foo` that will be quite heavy. I'd like to ignore all folders under `**/foo`

### Why is this important to you?

See description. I have many many monorepos of this style. 

### How are you currently working around this?

Sending 4.5 GB to the buildkit daemon build context 😅

**Comment by chakrit** [comment]
Instead of just the syntax. I think it makes more sense to just use the same docker's `.dockerignore` logic directly. This way any existing codebase with any docker-related toolings will have higher chance of just working correctly out of the box when switching to dagger.

**Comment by sheldonhull** [comment]
While defining ignore directly in dagger is great, I'd suggest that the same functionality where Dagger can define a build directly or use a DockerFile would be great here.

Since Dagger might be an incremental adoption for folks, setting `WithDockerIgnore(ignorefile)` as a method would be fantastic to allow incremental adoption while respecting the incremental adoption of Dagger.

**Comment by helderco** [comment]
> Sending 4.5 GB to the buildkit daemon build context 😅

The intended way to reduce the scope of the context is with the [`Include`/`Exclude` options](https://pkg.go.dev/dagger.io/dagger#HostDirectoryOpts) in [`Host.Directory`](https://pkg.go.dev/dagger.io/dagger#Host.Directory):

```go
source := client.Host().Directory(".", dagger.HostDirectoryOpts{
	Exclude: []string{"node_modules/", "ci/"},
})
```

[Full example](https://docs.dagger.io/947391/quickstart-test#test-the-application).

Notice this isn't only useful for building from a `Dockerfile`. If you don't filter your source directory in any other call, a minor unrelated change can bust your cache easily.

**Comment by marcosnils** [comment]
:wave: just published a helper function to deal with this in my Golang `dagger-libs` project: https://github.com/marcosnils/dagger-libs/blob/main/ddocker/utils.go. 

Here's an exmaple on how to use it: 

```go
	appDir := client.Host().Directory(".", dagger.HostDirectoryOpts{
		Exclude: ddocker.DockerIngoreExclude(".dockerignore"),
	})
```
